keras/backend/tensorflow_backend.py

```diff
@@ -808,6 +808,9 @@ def zeros_like(x, dtype=None, name=None):
     ```
     {{np_implementation}}
     """
+    if dtype is None:
+        dtype = floatx()
+    tf_dtype = tf.as_dtype(dtype)
```

Comment:
In modern TF you don't need this cast anymore (I've removed this pattern in the `tf-2` branch).